src/go/docker-credential-none/cmd/root.go

```diff
@@ -107,3 +116,50 @@ func saveParsedConfig(config *dockerConfigType) error {
 	}
 	return os.Rename(scratchFile.Name(), configFile)
 }
+
+/**
+ * Make the error-messages less cobra-like. In particular, if help is requested, let cobra deal with it.
+ * If there are any other non-help options, complain with the usage string.
+ * If extra arguments are given, complain with the usage string.
+ * If no subcommand is given, complain with the usage string.
+ * If an unrecognized subcommand is given, complain with the "Unknown credential action" message.
+ */
+func validateArgs() string {
+	const usage = "Usage: /usr/local/bin/docker-credential-none <store|get|erase|list>"
+	if len(os.Args) == 1 {
+		return usage
+	}
+	// Ignore help options
+	var commandName string
+	requestedHelp := false
+	returnUsageIfHelpNotRequested := func() string {
+		if requestedHelp {
+			return ""
+		}
+		return usage
+	}
+
+	for _, arg := range os.Args[1:] {
+		if arg[0] != '-' {
```

Comment:
`docker-credential-none ''` panics (because empty arg).

Reply:
The downside of not using a library... Good catch.